test/actions/katello/activation_key_test.rb

```diff
@@ -70,12 +70,23 @@ class UpdateWithoutCandlepinTest < TestBase
   class DestroyTest < TestBase
     let(:action_class) { ::Actions::Katello::ActivationKey::Destroy }
 
+    it 'does not plan with missing cp_id' do
+      action = create_action(action_class)
+      action.expects(:plan_self)
+      action.expects(:action_subject).with(activation_key)
+      plan_action(action, activation_key)
+      refute_action_planed(action, ::Actions::Candlepin::ActivationKey::Destroy)
+      assert_equal(activation_key.cp_id, nil)
+    end
+
     it 'plans' do
       action = create_action(action_class)
+      activation_key.update!(cp_id: 'something')
```

Comment:
The way you changed the test makes me think that the value of cp_id is significant since you later assert that's what the value is, but it doesn't seem like that's the case since the test doesn't actually _use_ that value anywhere

Two ways to (imo) improve this test:
- either refute_nil activation_key.cp_id (ie this is a precondition of this test)
- or assert that ::Actions::Candlepin::ActivationKey::Destroy was planned with a particular cp_id - the one from the activation key

Reply:
A present `activation_key.cp_id` is required to schedule the plan with this change:
```
plan_action(Candlepin::ActivationKey::Destroy, cp_id: activation_key.cp_id) if !skip_candlepin && activation_key.cp_id.present?
```

Reply:
OK, in that case I think the path forward is:

- keep the update! call as-is
- instead of asserting down below that the cp_id is the value you set, rather test the params given to the `::Actions::Candlepin::ActivationKey::Destroy` action reflect the cp_id (tons of examples of this in the test code)